mlflow/__init__.py

```diff
@@ -27,15 +27,35 @@
 
 For a lower level API, see the :py:mod:`mlflow.tracking` module.
 """
-from mlflow.version import VERSION as __version__  # pylint: disable=unused-import
-from mlflow.utils.logging_utils import _configure_mlflow_loggers
-import mlflow.tracking._model_registry.fluent
-import mlflow.tracking.fluent
+import google.protobuf
+from packaging.version import Version
+
+# This is a patch for protobuf <= 3.6.1, to make mlflow compatible with protobuf <= 3.6.1
+if Version(google.protobuf.__version__) <= Version("3.6.1"):
+    from google.protobuf.internal import enum_type_wrapper
+
+    # For version less than 3.6.1, Add EnumTypeWrapper.__getattr__ to access values
+    # see https://github.com/protocolbuffers/protobuf/pull/5234
```

Comment:
I'm really surprised __getattr__ was only supported since 2019 on OSS protobuf. In this case, I'd recommend to follow Corey's suggestion to require a minimum protobuf version instead of patching the EnumTypeWrapper class, which will miss all future updates on this function from the protobuf team.

For example, the patched function below is using the 3.6.1 implementation, which has since been improved:
https://github.com/protocolbuffers/protobuf/blob/master/python/google/protobuf/internal/enum_type_wrapper.py#L108